powerauth-push-server/src/test/io/getlime/push/client/PushServerClientTest.java

```diff
@@ -0,0 +1,207 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Class used for testing client-server methods
+ *
+ * @author Martin Tupy, martin.tupy.work@gmail.com
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(locations = "classpath:application-test.properties")
+public class PushServerClientTest {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @LocalServerPort
+    private int port;
+
+    @Test
+    public void getServiceStatusTest() throws Exception {
+        PushServerClient pushServerClient = new PushServerClient("http://localhost:" + port);
+        ObjectResponse<ServiceStatusResponse> actual = pushServerClient.getServiceStatus();
+        String body = restTemplate.getForEntity("http://localhost:" + port + "/push/service/status", String.class).getBody();
+        ObjectResponse<ServiceStatusResponse> expected = new ObjectMapper().readValue(body, new TypeReference<ObjectResponse<ServiceStatusResponse>>() {
+        });
+        assertThat(actual.getStatus()).isEqualTo(expected.getStatus());
+        assertThat(actual.getResponseObject().getApplicationDisplayName()).isEqualTo(expected.getResponseObject().getApplicationDisplayName());
+        assertThat(actual.getResponseObject().getApplicationEnvironment()).isEqualTo(expected.getResponseObject().getApplicationEnvironment());
+        assertThat(actual.getResponseObject().getApplicationName()).isEqualTo(expected.getResponseObject().getApplicationName());
+        // Time on every call is different can't be compared
+        // Assertions.assertThat(actual.getResponseObject().getTimestamp()).isEqualTo(parsedExpected.getResponseObject().getTimestamp());
+    }
+
+    @Test
+    public void createDeviceTest() throws Exception {
+        PushServerClient pushServerClient = new PushServerClient("http://localhost:" + port);
+        CreateDeviceRequest createDeviceRequest = new CreateDeviceRequest();
+        createDeviceRequest.setAppId(10L);
+        createDeviceRequest.setToken("1234567890987654321234567890");
+        createDeviceRequest.setPlatform(MobilePlatform.iOS.value());
+//        createDeviceRequest.setActivationId("49414e31-f3df-4cea-87e6-f214ca3b8412");
+        boolean actual1 = pushServerClient.createDevice(createDeviceRequest.getAppId(), createDeviceRequest.getToken(), MobilePlatform.iOS);
+//        boolean actual2 = pushServerClient.createDevice(createDeviceRequest.getAppId(), createDeviceRequest.getToken(), MobilePlatform.iOS, createDeviceRequest.getActivationId());
```

Comment:
Commented out code can go to repository only if it is part of some large functional change and will be used later. This is not the case - please do not commit commented out code.